### PR TITLE
Provide default content for blank project About pages (research, faq, etc...)

### LINKED ISSUE #2060: Provide default content for a project About page (research, faq, education, etc) if it does not exist
## Package
app-project

## Is your feature request related to a problem? Please describe.
If a project does not have any content for its About pages, that page should be served some default content so that the layout still renders. As suggested in PR for initial styling of new project About pages: https://github.com/zooniverse/front-end-monorepo/pull/2056#pullrequestreview-606735873

## Describe the solution you'd like
If the project About page does not exist because the team did not add any content, we could check for a 404 and load up the fallback for a missing page.

### PR DESCRIPTION
**Pull request by goplayoutside3**
Package:
app-project

Closes #2060.

Describe your changes:

- If a zooniverse project is created, and the owner doesn't click on 'About' in the project creator, no data is passed to the ProjectAboutPageConnector.
- This PR provides some default data for those cases so that the project's page layout still renders.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Review by srallen — COMMENTED** [review]
There are a few scenarios where content could be missing:
- The about pages are missing content entirely, which is currently coded for
- There are specific missing about pages, but not all of them. This currently is what's happening with the kitty test project, which is missing the team page, for example: http://localhost:3000/projects/brooke/i-fancy-cats/about/team
- Out of the pages that do exist for the kitty project, only the research page has content, while the rest have `content: null` and the fallback doesn't render for these. This is because the `content` object property exists, even though it is null for these pages, and so the default will not be used. 

Checking the current behavior, I made a quick test project, and a research page and a team page were created with default content for those: https://master.pfe-preview.zooniverse.org/projects/srallen086/untitled-project-4-28-2021-4-45-04-pm/about/research

The other pages, if they have `null` content, are removed from the link list, referencing both my new test project and the kitty test project: https://master.pfe-preview.zooniverse.org/projects/brooke/i-fancy-cats/about/research 

To summarize:
- Research and Team pages are required and have default content strings.
- Other pages are not required and if their content is null, should not be linkable.

I think it'd be good to confirm what the expected behaviors should be here before writing specs and making changes, so let's ask about it on slack.

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js [hunk 1]
```diff
@@ -1,18 +1,70 @@
 import { shallow } from 'enzyme'
-import React from 'react'
+import { ProjectAboutPage, PageHeading } from './ProjectAboutPage'
+import AboutDropdownNav from './components/AboutDropdownNav'
+import AboutSidebar from './components/AboutSidebar'
+import TeamMember from './components/TeamMember'
 
-import ProjectAboutPage from './ProjectAboutPage'
+let wrapper
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js:7]: We've learned this the hard way, but it's best to declare this within the describe scope rather than as a global. There can be unintended side effects otherwise with test running, if say someone forgets to declare wrapper and start using it and there's this global reference still hanging around.

There's a global leak detection we could enable for mocha since I think we still have a few of these around... and that's likely not helpful for you if you come across it as a reference!

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPageConnector.spec.js [hunk 2]
```diff
@@ -5,36 +5,105 @@ import sinon from 'sinon'
 import ProjectAboutPageConnector from './ProjectAboutPageConnector'
 import ProjectAboutPage from './ProjectAboutPage'
 
-describe('Component > ProjectAboutPageConnector', function () {
+describe('Component > ProjectAboutPageConnector', () => {
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPageConnector.spec.js:8]: Non-blocking comment, and just a FYI in case you weren't aware: [arrow functions are discouraged](https://mochajs.org/#arrow-functions), but depends on if you need mocha's context. We tend to default using `function() {}` for this reason just for safety.

File: packages/app-project/src/screens/ProjectAboutPage/components/TeamMember.js [hunk 3]
```diff
@@ -80,9 +80,9 @@ const TeamMember = ({ user, router }) => {
         />
         {user.roles &&
           user.roles.length &&
-          user.roles.map(role => (
+          user.roles.map((role, index) => (
             <StyledRole
-              key={role}
+              key={`${role}${index}`}
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/components/TeamMember.js:85]: Hm, I thought the role resource would have a unique id, perhaps we could use here? I'm not sure exactly if that's being preserved from the original API response by the time it gets here though?

File: packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.spec.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    }
+  }
+
+  before(() => {
+    wrapper = shallow(<AboutDropdownNav aboutNavLinks={[]} router={router} />)
+  })
+
+  it('should render without crashing', () => {
+    expect(wrapper).to.be.ok()
+  })
```
Comment (srallen) [inline packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.spec.js:20]: Could we expand the tests here to test for the conditional rendering happening in the dropdown nav similar to what you have in the specs for `AboutSidebar`?

**Review by srallen — CHANGES_REQUESTED** [review]
I have a few minor comments mostly related to tests, but other than that, it's looking good!

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPage.spec.js:7]: That totally makes sense! I'll be sure to move the declaration before merging.

File: packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPageConnector.spec.js (hunk 2 above)
Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectAboutPage/ProjectAboutPageConnector.spec.js:8]: I did not know that, but I'll make sure to avoid them in the future!

File: packages/app-project/src/screens/ProjectAboutPage/components/TeamMember.js (hunk 3 above)
Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectAboutPage/components/TeamMember.js:85]: The user object for each team member does have an id, but at this point `role` is a string. You're right though, it should be unique and index is not needed.

File: packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.spec.js (hunk 4 above)
Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.spec.js:20]: @srallen Because the dropdown uses `DropButton` from `grommet`, the `AboutNavLink` components in the `dropContent` are not rendered until a user clicks the drop button. I tried a few techniques like simulating a click or using `setProps` in the spec file, but couldn't get the menu to 'open' in tests.

Anything you recommend trying with `enzyme` to get the dropdown menu to render?

Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectAboutPage/components/AboutDropdownNav.spec.js:21]: Per our convo this morning, I separated the dropdown content into its own component for testing conditional rendering of the nav links 👍

**Review by srallen — APPROVED** [review]
LGTM. Thanks!
